pymc_bart/utils.py

```diff
@@ -2,17 +2,29 @@
 
 import warnings
 
+import pytensor.tensor as pt
 import arviz as az
 import matplotlib.pyplot as plt
 import numpy as np
-
+import numpy.typing as npt
 from pytensor.tensor.var import Variable
 from scipy.interpolate import griddata
 from scipy.signal import savgol_filter
-from scipy.stats import pearsonr, norm
+from scipy.stats import norm, pearsonr
+from typing import List, Optional, Tuple, Union
+from .tree import Tree
+
+
+TensorLike = Union[npt.NDArray[np.float_], pt.TensorVariable]
 
 
-def _sample_posterior(all_trees, X, rng, size=None, excluded=None):
+def _sample_posterior(
+    all_trees: List[List[Tree]],
```

Comment:
I am not sure what the type of `all_trees` should be. In the `_sample_posterior` function we use it as ` shape = stacked_trees[0][0].predict(X[0]).size` so I tried with `List[List[Tree]]` however the `predict` method from the `Tree` class returns a `float` and that is why `mypy` complains about:
```python
pymc_bart/utils.py:58: error: "float" has no attribute "size" 
```

Reply:
Predict should return an array with the number of elements determined by the first dimension of the shape argument passed to BART, so by default is only one element. But if we are fitting more than one BART variable like in the quartile regression example of the heteroscedasticity example this will have more elements